db/column_family.cc

```diff
@@ -1118,22 +1116,19 @@ SuperVersion* ColumnFamilyData::GetThreadLocalSuperVersion(
   if (sv == SuperVersion::kSVObsolete ||
       sv->version_number != super_version_number_.load()) {
     RecordTick(ioptions_.statistics, NUMBER_SUPERVERSION_ACQUIRES);
-    SuperVersion* sv_to_delete = nullptr;
 
     if (sv && sv->Unref()) {
       RecordTick(ioptions_.statistics, NUMBER_SUPERVERSION_CLEANUPS);
-      db_mutex->Lock();
+      db->mutex()->Lock();
       // NOTE: underlying resources held by superversion (sst files) might
       // not be released until the next background job.
-      sv->Cleanup();
-      sv_to_delete = sv;
+      db->AddSuperVersionsToFreeQueue(sv);
+      db->SchedulePurge();
```

Comment:
Right now we only do it with option avoid_unnecessary_blocking_io is true. Can we keep this pattern? We may not have the information inside class ColumnFamilyData, but I think ImmutableDBOptions is passed into the constructor, so we an add a member variable of ColumnFamilyData `bool avoid_unnecessary_blocking_io` to remember it.